docs-starlight/src/content/docs/05-troubleshooting/03-performance.mdx

```diff
@@ -0,0 +1,149 @@
+
+See [#1549](https://github.com/opentofu/opentofu/issues/1549) for more details.
+
+### Skip Dependency Inputs
+
+A rarely used capability of Terragrunt dependency blocks is the ability to read inputs from dependencies. There's very little in the way of valid use-cases for this, and supporting this behavior is a source of significant performance penalty for Terragrunt users.
+
+The reason for this is that Terragrunt needs to recursively read inputs from dependencies, and their dependencies, all the way down to the ancestral Terragrunt unit they all have in common to be able to expose the inputs to the dependent unit.
+
+You can mitigate this performance penalty by using the `skip-dependency-inputs` strict control.
+
+Enabling this strict control will enable a breaking change in Terragrunt behavior, making it so that dependency blocks no longer read inputs from dependencies.
+
+In the future, we will be removing this capability from Terragrunt, so you won't need to use this strict control.
+
+### Skip Dependency Inputs - Gotchas
+
+In general, you should use this strict control. The only reason not to use it is if you are using dependency blocks to read inputs from dependencies, which you should stop doing as soon as possible.
+
+## Measuring Performance
+
+Before diving into any particular performance optimization, it's important to first measure performance, and to make sure that you measure performance after any changes so that you understand the impact of your changes.
+
+To measure performance, you can use multiple tools, depending on your role.
+
+### End User
+
+As an end user, you're advised to use the following tools to get a better understanding of the performance of Terragrunt.
+
+#### OpenTelemetry
+
+Use [OpenTelemetry](/docs/troubleshooting/open-telemetry) to collect traces from Terragrunt runs so that you can analyze the performance of individual operations when using Terragrunt.
+
+This can be useful both to identify bottlenecks in Terragrunt, and to understand when performance changes can be attributed to integrations with other tools, like OpenTofu or Terraform.
+
+#### Benchmark Usage
+
+Use benchmarking tools like [Hyperfine](https://github.com/sharkdp/hyperfine) to run benchmarks of your Terragrunt usage to compare the performance of different versions of Terragrunt, or with different configurations.
+
+You can use configurations like the `--warmup` flag to do some warmup runs before the actual benchmarking. This is useful to get a more accurate measurement of the performance of Terragrunt with cache populated, etc.
+
+Here's an example of how to use Hyperfine to benchmark the performance of Terragrunt with two different configurations:
+
+```shell
+hyperfine -w 3 -r 5 'terragrunt run-all plan' 'terragrunt run-all plan --dependency-fetch-output-from-state'
+```
+
+### Developer
```

Comment:
This is for _terragrunt_ developers, right?  Maybe worth adding that qualifier, I was confused for a minute why these docs were teaching me about golang benchmarking

Reply:
Updated.